_posts/2024-03-20-strimzi-kraft-migration.md

```diff
@@ -0,0 +1,233 @@
+
+* `migration` to start the migration process;
+* `rollback` to revert the migration itself;
+
+During migration or rollback, the `enabled` and `disabled` values are still used to finalize the procedure by ending with a cluster in KRaft mode or keeping the ZooKeeper mode.
+
+One of the main prerequisites for the migration is about having the ZooKeeper-based cluster using the `KafkaNodePool`(s) to run the brokers.
+If that is not the case for you, please refer to the official Strimzi documentation [here](https://strimzi.io/docs/operators/latest/deploying#proc-migrating-clusters-node-pools-str).
+
+Following there is a snippet of a `Kafka` custom resource highlighting node pools enabled and using ZooKeeper for metadata.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: my-cluster
+  annotations:
+    strimzi.io/kraft: disabled # states that the cluster is ZooKeeper-based
+    strimzi.io/node-pools: enabled
+# ...
+```
+
+The deployed cluster has severals pods running for the Kafka brokers and ZooKeeper nodes.
+
+```shell
+NAME                                          READY   STATUS    RESTARTS   AGE
+my-cluster-entity-operator-555785d699-kksrs   2/2     Running   0          17m
+my-cluster-kafka-0                            1/1     Running   0          18m
+my-cluster-kafka-1                            1/1     Running   0          18m
+my-cluster-kafka-2                            1/1     Running   0          18m
+my-cluster-zookeeper-0                        1/1     Running   0          20m
+my-cluster-zookeeper-1                        1/1     Running   0          20m
+my-cluster-zookeeper-2                        1/1     Running   0          20m
+strimzi-cluster-operator-7ddf57685d-245pd     1/1     Running   0          17m
+```
+
+#### From ZooKeeper to full KRaft based brokers
+
+The first step is to deploy a `KafkaNodePool` custom resource which hosts the KRaft controllers and then changing the `strimzi.io/kraft` annotation from `disabled` to `migration` to trigger the process.
+
+Following a snippet of a `KafkaNodePool` custom resource to be deployed for getting the KRaft controllers running.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaNodePool
+metadata:
+  name: controller
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  replicas: 3
+  roles:
+    - controller
+  storage:
+    type: jbod
+    volumes:
+      - id: 0
+        type: persistent-claim
+        size: 100Gi
+        deleteClaim: false
+```
+
+The corresponding `Kafka` custom resource is then updated in regards to the `strimzi.io/kraft` set to `migration`. 
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: my-cluster
+  annotations:
+    strimzi.io/kraft: migration # updated from "disabled" to "migration" to trigger the migration process
+    strimzi.io/node-pools: enabled
+# ...
+```
+
+The annotation change triggers the Strimzi operator to run the following steps for you:
```

Comment:
```suggestion
The annotation change is detected by the Strimzi operator which runs the following steps:
```

Reply:
Tbh just removed "for you" but I like the current statement.